packages/browser/src/plugins/in-app-plugin/index.ts

```diff
@@ -0,0 +1,149 @@
+                Gist.events.on(event, (message: any) => {
+                    _eventTarget.dispatchEvent(newEvent(gistToCIO(event), { 
+                        messageId: message.messageId,
+                        deliveryId: message.properties?.gist?.campaignId,
+                    }));
+                });
+            });
+        }
+
+        Gist.events.on('messageShown', (message: any) => {
+            const deliveryId:string = message?.properties?.gist?.campaignId;
+            if (typeof deliveryId != 'undefined' && deliveryId != '') {
+                _analytics.track(semanticEvents.JourneyMetric, {
+                    'deliveryId': deliveryId,
+                    'metric': semanticEvents.Opened,
+                });
+            }
+        });
+
+        Gist.events.on('messageAction', (params: any) => {
+            const deliveryId:string = params?.message?.properties?.gist?.campaignId;
+            if (params.action != 'gist://close' && typeof deliveryId != 'undefined' && deliveryId != '') {
+                _analytics.track(semanticEvents.JourneyMetric, {
+                    'deliveryId': deliveryId,
+                    'metric': semanticEvents.Clicked,
+                    'actionName': params.name,
+                    'actionValue': params.action,
+                });
+            }
+            settings.events && _eventTarget.dispatchEvent(newEvent(InAppEvents.MessageAction, {
+                messageId: params.message.messageId,
+                deliveryId: deliveryId,
+                action: params.action,
+                name: params.name,
+                actionName: params.name,
+                actionValue: params.action,
+                message:{
+                    dismiss: function() {
+                        Gist.dismissMessage(params?.message?.instanceId);
+                    }
+                }
+             }));
+        });
+    }
+
+    async function page(ctx: Context): Promise<Context> {
+        if(!_pluginLoaded)
+            return ctx;
+
+        let page:string = ctx.event?.properties?.name || ctx.event?.properties?.url;
+        if(page != "") {
+            Gist.setCurrentRoute(page);
+        }
+
+        return ctx;
+    }
+
+    async function reset(ctx: Context): Promise<Context> {
+        await Gist.clearUserToken();
+        return ctx;
+    }
+    
+    async function syncUserToken(ctx: Context): Promise<Context> {
+        if(!_gistLoaded)
+            return ctx;
+
+        let user = _analytics.user().id();
+        if (typeof user !== 'undefined' && user != null && user != "") {
+            await Gist.setUserToken(user);
+        } else {
+            await Gist.clearUserToken();
+        }
+        return ctx;
+    }
+
+    const customerio: Plugin = {
+        name: 'Customer.io In-App Plugin',
+        type: 'before',
+        version: '0.0.1',
+        isLoaded: (): boolean => _pluginLoaded,
+        load: async (ctx: Context, instance: Analytics) => {
```

Comment:
Can you also implement unload and unregister the event listeners? otherwise they'll leak if a user unloads the plugin.